src/context/AppProvider/AppContext.tsx

```diff
@@ -114,31 +118,56 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useEffect(() => {
     if (!wallet) return
     ;(async () => {
-      try {
-        const accountNumber = 0
-        const chainIds = supportedChains
-        const accountMetadataByAccountId = await deriveAccountIdsAndMetadata({
-          accountNumber,
-          chainIds,
-          wallet,
+      let chainIds = Array.from(supportedChains).filter(chainId =>
+        walletSupportsChain({ chainId, wallet }),
+      )
+      const isMultiAccountWallet = wallet.supportsBip44Accounts()
+      for (let accountNumber = 0; chainIds.length > 0; accountNumber++) {
+        // only some wallets support multi account
+        if (accountNumber > 0 && !isMultiAccountWallet) break
+        const input = { accountNumber, chainIds, wallet }
+        const accountMetadataByAccountId = await deriveAccountIdsAndMetadata(input)
+        const accountIds: AccountId[] = Object.keys(accountMetadataByAccountId)
+        const { getAccount } = portfolioApi.endpoints
+        const opts = { forceRefetch: true }
```

Comment:
Started using this naming pattern for RTK query options but obviously fine
```suggestion
        const queryOptions: StartQueryActionCreatorOptions = { forceRefetch: true }
```

Reply:
ur ruining my 1 line golf